Document Ice 3.6.0 version requirements, installation instructions and CI changes

rleigh-dundee opened the pull request:
To match the changes in https://github.com/openmicroscopy/openmicroscopy/pull/3911 and pending CI changes.

Commit:
version-requirements: Update for Ice 3.6

Note will require further updates as homebrew and the various
distributions pick up Ice 3.6.

Commit:
omero: Update installation instructions for Ice 3.6

Commit:
contributing: Update ci-omero for Ice 3.6

mtbc commented:
For what it's worth, I don't see any typos.

jburel commented:
These jobs builds => should be build

Commit:
contributing: Correct plurals in ci-omero

joshmoore commented on omero/sysadmins/version-requirements.txt:
```diff
@@ -160,7 +160,7 @@ Components
 * Ice
 
-  * [5.1] 3.4 deprecated, 3.5 recommended, 3.6 not supported
+  * [5.1] 3.4 deprecated, 3.5 recommended, 3.6 supported
```
This "supported" may end up needing to be "experimental" depending on how far we get with the rest of https://trello.com/c/OHLViGl8/47-ice-3-6

hflynn commented:
I assume this is also on hold given code PR is now against decoupling.

hflynn commented:
Do we need to close this PR as Ice 3.6 work is now shunted to 5.2?

manics commented:
I thought it was still going to be experimental?

rleigh-dundee commented:
Yes, I was quite surprised it got pushed to 5.2.  It's contrary to what we discussed, and will leave current homebrew users and Windows users without an officially supported Ice.

hflynn commented:
@joshmoore what is the plan here?

joshmoore commented:
"Support" for 3.6 is 5.2 along with dropping 3.4. An experimental PR against decoupling was still the plan, @rleigh-dundee, no? (https://github.com/openmicroscopy/openmicroscopy/pull/3911#issuecomment-116646134)

hflynn commented:
Ok, so we are going to have an experimental download rather than claiming 'support' for 5.1.3 - what do you want to do about this PR @rleigh-dundee ? We could leave it for 5.1.4/5.2 and just cover the artifacts being available in the release announcement, or you could amend this as needed?

manics commented:
How about keep the ci bit, and condense all the 3.6 experimental stuff (including caveats/warnings) into a single section that can be linked from the release announcement?

hflynn commented:
That would work for me

rleigh-dundee commented:
I'll update it to be experimental.

rleigh-codelibre committed:
omero: Ice 3.6 is not supported for 5.1.3

rleigh-codelibre committed:
version-requirements: Update Ice 3.6 support status

rleigh-codelibre committed:
version-requirements: Update for current releases

rleigh-dundee commented:
OK, I updated this to use "upcoming" rather than "experimental" because that's the nomenclature we were already using on our version-requirements page for this specific category of support (new, not yet fully supported or tested, use at own risk).

I also updated the version requirements page at the same time since I noticed it was missing six months worth of releases.  This adds Ubuntu 15.04, Debian 8.0, Fedora 22 and adds Ubuntu 15.10, Debian 9.0 and Fedora 23 as upcoming versions, and adjusts the support state for new releases of each of our dependencies for which there have been end-of-life or new releases (e.g. Ubuntu 14.10 dropped since its end of life is next week).

manics commented on omero/sysadmins/unix/server-installation.txt:
```diff
@@ -248,22 +248,23 @@ Ice
++-----------+-------------------------------+
+| Debian    | zeroc-ice                     |
++-----------+-------------------------------+
+| Homebrew  | ice (with --with-java option) |
```
Don't you still need `--with-python`? And for 5.1should this be `zeroc-ice` from the `ome` tap?

rleigh-dundee replied:
This is documenting the current state of upstream homebrew.  The `ome` tap is what the separate homebrew page is about, and is deliberately omitted here.  So long as we've adjusted our omero formula in the tap to use the ice from the tap also, that should be covered with no documentation changes required.

hflynn replied:
FYI - this is rendering as '-with-java option' on staging, you might need to use back tick mark-up to get the double dash i think

hflynn commented on omero/sysadmins/version-requirements.txt:
```diff
@@ -1425,12 +1456,12 @@ Ice
+      - Upcoming
       - Recommended
-      -
+      - `Ice 3.6.0 <http://zeroc.com/forums/announcements/6468-ice-3-6-0-ice-touch-3-6-0-released.html>`__
```
https://forums.zeroc.com/forum/announcements/6631-ice-3-6-0-and-ice-touch-3-6-0-released
Your link is just redirecting to the top level of the forum

rleigh-dundee replied:
Looks like they changed their forum software and it broke all the links.

manics commented on omero/sysadmins/unix/server-installation.txt:
```diff
@@ -248,22 +248,23 @@ Ice
++-----------+-------------------------------+
 
 The Ice version may vary, depending upon the distribution version you
 are using. The Ice versions in currently supported versions of Debian
```
`are shown in the :ref:ice-requirements of the :doc:/sysadmins/version-requirements page` (a couple lines below) comes out as `are shown in the Ice of the Version requirements page.`

manics commented on omero/sysadmins/unix/server-installation.txt:
```diff
@@ -283,6 +284,17 @@ latest release is available from the :zeroc:`ZeroC website
+    With Ice 3.6, the Python bindings are provided separately.  If
+    your package manager does not provide Ice python packages, run::
+
+        pip install zeroc-ice
```
This is displayed in sphinx with yellow/green side bars. Doesn't bother me, @hflynn ?

rleigh-dundee replied:
If anyone knows CSS, it would be nice to fix the broken stuff we've been using.

hflynn replied:
I think you can get rid of it if you reduce the indentation a bit

rleigh-dundee replied:
Tried and it didn't work, so rephrased.

manics commented on omero/sysadmins/windows/server-installation.txt:
```diff
@@ -146,6 +128,40 @@ The following are optional depending on what services you require:
+    (recommended) you should use the versions specified below. In
+    particular since OMERO does not support Python 3 **it will not
+    work with the ZeroC Ice 3.5 package for Windows**. However, Ice
+    3.5 will work when built against Python 2.7.  However, the ZeroC
```
Two consecutive sentences starting with `However`

hflynn commented on omero/sysadmins/windows/server-installation.txt:
```diff
@@ -83,11 +62,14 @@ Python 2
-- If you have downloaded Ice 3.5.1 from our downloads page, you will
-  need to install the `latest Python 2.7 release <http://www.python.org/downloads/windows/>`_.
+- If you have downloaded Ice 3.5.1 from our downloads page or Ice
+  3.6.0 from ZeroC (note that support is currently upcoming), you will
```
I know you explain 'upcoming' = experimental support in the versions requirements page but incase someone hasn't read that, it wouldn't hurt to have that stated explicitly here too.

hflynn commented on omero/sysadmins/windows/server-installation.txt:
```diff
@@ -146,6 +128,40 @@ The following are optional depending on what services you require:
+Ice 3.4 installers can be found on the old :zeroc:`ZeroC Ice 3.4.2
+download page <download_3_4_2.html>`.  Alternatively, the :zeroc:`Ice
+3.6.0 download page <download.html>` provides installers for Windows,
+but support for this release is currently upcoming.
```
Similarly, 'upcoming' may not be clear enough here.

Do you want to mention somewhere that the 3.6 server version is available from the artifactory given it won't be mentioned on the main downloads page as yet?

rleigh-dundee replied:
Unsure.  Given the level of support and testing, it might be best to leave it public but without any references, so anyone who explicitly asks can be pointed to them, but no one will use them by accident.

hflynn replied:
Fair enough, makes sense to me.

Commit:
omero: Correct ZeroC forum URLs (all invalidated)

Commit:
omero: server-installation: Mark up --with-java option

Commit:
omero: server-installation: Clarify meaning of 'upcoming'

Commit:
omero: server-installation: Remove overuse of 'however'

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-5.1-merge-push#1107](https://ci.openmicroscopy.org/job/OMERO-5.1-merge-push/1107/). See the [console output](https://ci.openmicroscopy.org/job/OMERO-5.1-merge-push/1107/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - omero/sysadmins/version-requirements.txt

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-5.1-merge-docs#583](https://ci.openmicroscopy.org/job/OMERO-5.1-merge-docs/583/). See the [console output](https://ci.openmicroscopy.org/job/OMERO-5.1-merge-docs/583/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - omero/sysadmins/version-requirements.txt

sbesson commented:
Can you merge `origin/develop` into this branch. ZeroC links had been reviewed as part of https://github.com/openmicroscopy/ome-documentation/pull/1237

rleigh-codelibre committed:
Merge remote-tracking branch 'origin/develop' into ice36

Conflicts:
	omero/sysadmins/version-requirements.txt

rleigh-dundee commented:
@sbesson Done.

hflynn commented:
Rebuilding on staging now

hflynn commented:
Looks sorted to me now, will be holding off merging for code and CI changes though.

joshmoore commented:
Excluding this for the moment since https://github.com/openmicroscopy/openmicroscopy/issues/3957 is ON_HOLD since it's hiding URL errors in the merge build.

hflynn commented:
I'll have to cherry-pick the ZeroC forum link fixes if none of the Ice 3.6 updates are going in for the release

jburel commented:
Closing. We can re-activate it when we add support for Ice 3.6
